stable/appmesh-manager/README.md

```diff
@@ -84,6 +86,92 @@ helm upgrade -i appmesh-manager eks/appmesh-manager \
 
 The [configuration](#configuration) section lists the parameters that can be configured during installation.
 
+
+## Upgrade
+
+
+### Upgrade without preserving old App Mesh resources
+
+```sh
+# Keep old App Mesh controller running, it is responsible to cleanup App Mesh resources in AWS
+# Delete all existing App Mesh custom resources (CRs)
+kubectl delete virtualservices --all --all-namespaces
+kubectl delete virtualnodes --all --all-namespaces
+kubectl delete meshes --all --all-namespaces
+
+# Delete all existing App Mesh CRDs
+kubectl delete customresourcedefinition/virtualservices.appmesh.k8s.aws
+kubectl delete customresourcedefinition/virtualnodes.appmesh.k8s.aws
+kubectl delete customresourcedefinition/meshes.appmesh.k8s.aws
+# Note: If a CRD stuck in deletion, it means there still exists some App Mesh custom resources, please check and delete them.
+
+# Delete App Mesh controller
+helm delete appmesh-controller -n appmesh-system
+
+# Delete App Mesh injector
+helm delete appmesh-inject -n appmesh-system
+```
+
+Run the `appmesh-manager/upgrade/pre_upgrade_check.sh` script and make sure it passes before you proceed
+
+Now you can proceed with the installation steps described above
+
+### Upgrade preserving old App Mesh resources
+
+```sh
+# Save manifests of all existing App Mesh custom resources
+kubectl get virtualservices --all-namespaces -o yaml > virtualservices.yaml
+kubectl get virtualnodes --all-namespaces -o yaml > virtualnodes.yaml
+kubectl get meshes --all-namespaces -o yaml > meshes.yaml
+
+# Delete App Mesh controller, so it won’t clean up App Mesh resources in AWS while we deleting App Mesh CRs later.
+helm delete appmesh-controller -n appmesh-system
+
+# Delete App Mesh injector.
+helm delete appmesh-inject -n appmesh-system
+
+# Remove finalizers from all existing App Mesh CRs. Otherwise, you won’t be able to delete them.
```

Comment:
there should be commands to remove finalizer from objects. we should be able to do a version by combine  `kubectl get` and `jq`.

I didn't code it originally due to time., but best to have it here :D

Reply:
That's a good idea. I will add it